LineSorter/Helpers/TextSelection.cs

```diff
@@ -51,7 +52,7 @@ public static void ReplaceSelection(this IEnumerable<string> Selections, EmptyLi
             string newlineStr = NewlineType.AsString();
             if (Action == EmptyLineAction.DependsOnSettings)
                 Action = VSPackage.Loader.Settings.EmptyLineAction;
-            DTE dte = ServiceProvider?.GetService(typeof(DTE)) as DTE;            
+            DTE2 dte = ServiceProvider?.GetService(typeof(DTE)) as DTE2;            
```

Comment:
Could you tell me more about this? I've skipped through the video you mentioned, but it seems that Mads is mostly talking about `InstallationTarget`, shared projects as a way for easier support of different VS versions, and so on. Didn't see anything `DTE -> DTE2`-related

Also, I'm looking at [DTE's declaration in VS 2022 SDK](https://docs.microsoft.com/en-us/dotnet/api/envdte.dte?view=visualstudiosdk-2022), and `ActiveDocument` and `ExecuteCommand` are still there. So why should we cast `DTE` to `DTE2` then?

Reply:
I'm sorry, I linked the wrong video. Here's the correct one, with a timestamp in the link where he demonstrates a case where DTE no longer works: https://youtu.be/-PKIPTW6km0?t=1138

The gist of it is that the inheritance of DTE has changed in VS2022, so many things are no longer available on DTE. However DTE2 contains all that DTE had, and more (and I believe it works in 2017, 2019 and 2022). So using DTE2 should always be a better choice.

Before changing to DTE2, I got an error when testing the extension, switching to DTE2 solved the problem.